src/go/rdctl/cmd/api.go

```diff
@@ -86,29 +94,31 @@ func doAPICommand(cmd *cobra.Command, args []string) error {
 			apiSettings.Method = "PUT"
 		}
 		if apiSettings.InputFile == "-" {
-			contents, err = ioutil.ReadAll(os.Stdin)
+			contents, err = io.ReadAll(os.Stdin)
 		} else {
-			contents, err = ioutil.ReadFile(apiSettings.InputFile)
+			contents, err = os.ReadFile(apiSettings.InputFile)
 		}
 		if err != nil {
 			return err
 		}
-		result, errorPacket, err = processRequestForAPI(doRequestWithPayload(apiSettings.Method, endpoint, bytes.NewBuffer(contents)))
+		response, err := rdClient.DoRequestWithPayload(apiSettings.Method, endpoint, bytes.NewBuffer(contents))
```

Comment:
I hate to say this, but it seems to be more idiomatic Go-ish to not pass a non-nil `err` to `ProcessRequestForAPI`, and instead do it this way:
```
response, err := rdClient.DoRequestWithPayload(...)
if err != nil {
  return nil, nil, err
}
result, errorPacket, err := client.ProcessRequestForAPI(response)
```

But if everyone on the team is ok with the current way of passing an error on rather than dealing with it right away, so am I. I'm the one who wrote the original code that way, but I only did that do make the client call a one-liner.

Reply:
I think the non-nested way of writing it is clearer (and more idiomatic).